core/station/impl/src/repositories/indexes/request_policy_resource_index.rs

```diff
@@ -158,6 +191,49 @@ impl RequestPolicyResourceIndexRepository {
             policies
         })
     }
+
+    // Find all external canister call policies related to the specified canister id and execution method.
+    pub fn find_external_canister_call_policies_by_execution_method(
+        &self,
+        canister_id: &Principal,
+        execution_method: &String,
+    ) -> Vec<UUID> {
+        DB.with(|db| {
+            db.borrow()
+                .range(
+                    (RequestPolicyResourceIndex {
+                        resource: Resource::ExternalCanister(ExternalCanisterResourceAction::Call(
+                            CallExternalCanisterResourceTarget {
+                                execution_method: ExecutionMethodResourceTarget::ExecutionMethod(
+                                    CanisterMethod {
+                                        canister_id: *canister_id,
+                                        method_name: execution_method.clone(),
+                                    },
+                                ),
+                                validation_method: ValidationMethodResourceTarget::No,
+                            },
+                        )),
+                        policy_id: [u8::MIN; 16],
+                    })..,
+                )
+                .take_while(|(index, _)| {
+                    matches!(
+                        &index.resource,
+                        Resource::ExternalCanister(ExternalCanisterResourceAction::Call(
+                            CallExternalCanisterResourceTarget {
+                                execution_method: ExecutionMethodResourceTarget::ExecutionMethod(
+                                    CanisterMethod { canister_id: id, method_name: method }
+                                ),
+                                ..
+                            }
+                        ))
+                        if id == canister_id && execution_method == method
+                    )
+                })
```

Comment:
Isn't this fragile? It seems to depend on the fact that `ValidationMethodResourceTarget::No` is the least value of the `ValidationMethodResourceTarget` type.

Reply:
It does depend on that, we use this in other places as well with range queries. To make this more resilient we could instead of deriving `[(PartialOrd, Ord)]` we could implement that manually, but i don't think we need to do it for now.

Reply:
> i don't think we need to do it for now

If we have a test, we should fine actually.